### PR TITLE
Remove unused desktop code

### PR DESCRIPTION
**Pull request by amacneil**
**User-Facing Changes**
Remove support for self-building the desktop app.

**Description**
The official Foxglove desktop app is a binary build that incorporates both open-source and closed-source features to enable robotics observability workflows (both local + cloud). 

This PR removes desktop code from studio repo, which is not used by our internal binary release process. Publishing a desktop app is complex, requires a signing certificate on macOS and Windows, and is not something we provide support for. Community members are welcome to maintain their own desktop derivatives, but we encourage use of our [official desktop binaries](https://github.com/foxglove/studio/releases) or our [open source web builds](https://github.com/orgs/foxglove/packages/container/package/studio) going forward.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,19 +16,11 @@
...
-$ yarn desktop:start        # launch electron
-
-# To launch the browser app:
+# To launch the app:
 $ yarn web:serve
 
 # To launch the storybook:
 $ yarn storybook
-
-# Advanced usage: running webpack and electron on different computers (or VMs) on the same network
```
Comment (jtbandes) [inline CONTRIBUTING.md:29]: Did these instructions get kept somewhere?

**Review by jtbandes — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (amacneil) [reply CONTRIBUTING.md:29]: 👍 will migrate

Comment (jtbandes) [reply CONTRIBUTING.md:29]: @snosenzo  will need to migrate this PR to the internal repo https://github.com/foxglove/studio/pull/6941

File: package.json [hunk 2]
```diff
@@ -40,22 +35,18 @@
     "test": "jest",
     "test:debug": "cross-env NODE_OPTIONS='--inspect-brk' jest",
     "test:watch": "jest --watch",
-    "test:integration": "jest --forceExit --projects desktop/integration-test",
     "test:web-integration": "jest --forceExit --projects  web/integration-test",
     "test:web-integration:local": "BASE_URL=\"http://localhost:8080/\" jest --forceExit --projects  web/integration-test",
     "package:win": "yarn package --win",
     "package:darwin": "yarn package --mac",
     "package:linux": "yarn package --linux",
```
Comment (jtbandes) [inline package.json:42]: missed a few

**Pull request merged**